src/west/commands/project.py

```diff
@@ -819,6 +819,16 @@ def _ref_ok(project, ref):
     return _git(project, 'show-ref --quiet --verify ' + ref, check=False) \
            .returncode == 0
 
+def _head_ok(project):
+    # Returns True if the reference 'HEAD' exists and is not a tag/branch.
```

Comment:
Maybe this could say "...and is not a tag or remote ref (e.g. refs/remotes/origin/HEAD)", just to be a bit clearer.

Reply:
will do

Reply:
updated